src/adaptors/tcp/tcp_adaptor.c

```diff
@@ -2051,17 +2046,19 @@ static void on_accept(qd_adaptor_listener_t *adaptor_listener, pn_listener_t *pn
     conn->context.context = conn;
     conn->context.handler = on_connection_event_LSIDE_IO;
 
+    conn->raw_conn = pn_raw_connection();
+    pn_raw_connection_set_context(conn->raw_conn, &conn->context);
+    if (listener->protocol_observer) {
+        conn->observer_handle = qdpo_begin(listener->protocol_observer, conn->common.vflow, conn, conn->conn_id);
+    }
+
     sys_mutex_lock(&listener->lock);
+    DEQ_INSERT_TAIL(listener->connections, conn);
```

Comment:
Can the tcpListener delete loop be executed by the mgmt thread just before the `listener->lock` was taken here?   This new` conn` will not have the `management_close` flag set then so it will not be terminated.